examples/drive-thru/evals/test_agent.py

```diff
@@ -0,0 +1,222 @@
+from livekit.agents.llm.chat_context import ChatContext
+from livekit.agents.voice.run_result import mock_tools
+import pytest
+from livekit.agents import AgentSession, llm, ChatContext
+from livekit.plugins import openai
+
+from ..drivethru_agent import DriveThruAgent, new_userdata
+
+
+def _llm_model() -> llm.LLM:
+    return openai.LLM(model="gpt-4o", parallel_tool_calls=False, temperature=0.45)
+
+
+@pytest.mark.asyncio
+async def test_item_ordering() -> None:
+    userdata = await new_userdata()
+
+    async with (
+        _llm_model() as llm,
+        AgentSession(llm=llm, userdata=userdata) as sess,
+    ):
+        # add big mac
+        await sess.start(DriveThruAgent(userdata=userdata))
+        result = await sess.run(user_input="Can I get a Big Mac, no meal?")
+        result.expect.nth(0).is_function_call(
+            name="order_regular_item", arguments={"item_id": "big_mac"}
+        )
+        fnc_out = result.expect.nth(1).is_function_call_output()
+        assert fnc_out.event().item.output.startswith("The item was added")
+        result.expect.nth(2).is_message(role="assistant")
+
+        # remove item
+        result = await sess.run(user_input="No actually I don't want it")
+        try:
+            result.expect.nth(0).is_function_call(name="list_order_items")
+            result.expect.nth(1).is_function_call_output()
+            result.expect.nth(2).is_function_call(name="remove_order_item")
+            result.expect.nth(3).is_function_call_output()
+            result.expect.nth(4).is_message(role="assistant")
+        except AssertionError:
+            result.expect.nth(0).is_function_call(name="remove_order_item")
```

Comment:
this is the reason why I suggest that we shouldn't be too explicit about the ordering of calls and/or optional function calls like `list_order_items`